### PR TITLE
ESLint 9

### LINKED ISSUE #9496: Migration ESLint 9.0.0 (Angular React Vue apps)
https://eslint.org/docs/latest/use/configure/migration-guide

### PR DESCRIPTION
**Pull request by qmonmert**
Fix #9496

### CODE REVIEW

File: src/main/resources/generator/client/react/eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  {
+    languageOptions: {
+      globals: {
+        ...globals.node,
+      },
+    },
+  },
+  {
+    ignores: [
+      'node_modules/',
```
Comment (murdos) [inline src/main/resources/generator/client/react/eslint.config.js:16]: `node_modules` is already ignored by default, see https://eslint.org/docs/latest/use/configure/ignore#ignoring-files

File: src/main/java/tech/jhipster/lite/generator/client/vue/core/domain/VueModulesFactory.java [hunk 2]
```diff
@@ -55,7 +55,8 @@ public JHipsterModule buildVueModule(JHipsterModuleProperties properties) {
         .addDependency(packageName("vue-router"), VUE)
         .addDevDependency(packageName("@typescript-eslint/parser"), COMMON)
         .addDevDependency(packageName("@vitejs/plugin-vue"), VUE)
-        .addDevDependency(packageName("@vue/eslint-config-typescript"), VUE)
+        .addDevDependency(packageName("typescript-eslint"), COMMON)
+        .addDevDependency(packageName("globals"), COMMON)
         .addDevDependency(packageName("@vue/eslint-config-prettier"), VUE)
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/generator/client/vue/core/domain/VueModulesFactory.java:60]: This one could probably be removed, I think it's not used anymore:
```suggestion
```

File: src/main/java/tech/jhipster/lite/generator/client/vue/core/domain/VueModulesFactory.java [hunk 3]
```diff
@@ -55,7 +55,8 @@ public JHipsterModule buildVueModule(JHipsterModuleProperties properties) {
         .addDependency(packageName("vue-router"), VUE)
         .addDevDependency(packageName("@typescript-eslint/parser"), COMMON)
         .addDevDependency(packageName("@vitejs/plugin-vue"), VUE)
-        .addDevDependency(packageName("@vue/eslint-config-typescript"), VUE)
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/generator/client/vue/core/domain/VueModulesFactory.java:58]: Don't forget to remove this dependency (and `@vue/eslint-config-prettier`) from `src/main/resources/generator/dependencies/vue/package.json`

File: src/main/java/tech/jhipster/lite/generator/typescript/core/domain/TypescriptModuleFactory.java [hunk 4]
```diff
@@ -18,19 +18,20 @@ public JHipsterModule buildModule(JHipsterModuleProperties properties) {
...
     return moduleBuilder(properties)
       .packageJson()
+        .addType("module")
         .addDevDependency(packageName("typescript"), COMMON)
         .addDevDependency(packageName("@tsconfig/recommended"), COMMON)
         .addDevDependency(packageName("@typescript-eslint/eslint-plugin"), COMMON)
         .addDevDependency(packageName("@typescript-eslint/parser"), COMMON)
         .addDevDependency(packageName("@vitest/coverage-istanbul"), COMMON)
         .addDevDependency(packageName("eslint"), COMMON)
-        .addDevDependency(packageName("eslint-import-resolver-typescript"), COMMON)
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/generator/typescript/core/domain/TypescriptModuleFactory.java:27]: Don't forget to remove these now unused dependencies from `src/main/resources/generator/dependencies/common/package.json`

File: src/main/resources/generator/dependencies/common/package.json [hunk 5]
```diff
@@ -15,18 +15,19 @@
...
     "cssnano": "7.0.5",
     "cypress": "13.14.0",
-    "eslint": "8.57.0",
+    "eslint": "9.9.0",
     "eslint-config-prettier": "9.1.0",
-    "eslint-import-resolver-typescript": "3.6.3",
     "eslint-plugin-cypress": "3.5.0",
-    "eslint-plugin-import": "2.29.1",
+    "eslint-plugin-import-x": "4.1.0",
+    "eslint-plugin-prettier": "5.2.1",
```
Comment (murdos) [inline src/main/resources/generator/dependencies/common/package.json:29]: eslint-plugin-prettier is not used anymore

File: src/main/resources/generator/typescript/eslint.config.js.mustache [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    languageOptions: {
+      globals: {
+        ...globals.node,
+      },
+    },
+  },
+  {
+    ignores: [
+      'src/main/docker/',
+      'jest.conf.js',
```
Comment (murdos) [inline src/main/resources/generator/typescript/eslint.config.js.mustache:17]: Neither Jest, nor postcss is used.
Also why ignore `vitest.config.ts` and `eslint.config.js`?

**Comment by qmonmert** [comment]
thanks @murdos for your reviews

File: src/main/java/tech/jhipster/lite/generator/client/vue/core/domain/VueModulesFactory.java (hunk 3 above)
Comment (qmonmert) [reply src/main/java/tech/jhipster/lite/generator/client/vue/core/domain/VueModulesFactory.java:59]: done

**Review by murdos — APPROVED** [review]
LGTM. Could you fix conflict with the main branch?

**Comment by qmonmert** [comment]
@murdos thanks for the reviews

**Pull request merged**